packages/rpc-spec/src/rpc.ts

```diff
@@ -68,12 +67,12 @@ function createPendingRpcRequest<TRpcMethods, TRpcTransport extends RpcTransport
     return {
         async send(options?: RpcSendOptions): Promise<TResponse> {
             const { methodName, params, responseTransformer } = pendingRequest;
-            const payload = createRpcMessage(methodName, params);
-            const response = await rpcConfig.transport<RpcResponseData<unknown>>({
-                payload,
+            const response = await rpcConfig.transport<TResponse>({
+                payload: createRpcMessage(methodName, params),
                 signal: options?.abortSignal,
             });
-            return (responseTransformer ? responseTransformer(response, methodName) : response) as TResponse;
+            const responseData = await response.json();
+            return responseTransformer ? responseTransformer(responseData, methodName) : responseData;
```

Comment:
Don't you want to pass the `response` directly to the transformer here, to avoid calling `.json()` unless no transformer is provided?

Reply:
That comes in two PRs haha (spoiler alert).

Reply:
Yeppp this is what I figured haha.